### ISSUE TITLE
Data Caching

### ISSUE THREAD

**Issue by mtcdood** [churay/ytpa#58]
Without caching data this tool is basically unusable with Google's large reduction of default YouTube Data API quotas from 100,000 to 10,000. Loading a large channel can exceed half of the API quota available to a new API key, and Google is unlikely to increase quota before seeing a few days or even weeks of high quota usage to establish that the quota increase is actually necessary.

**Comment by churay** [comment]
I did notice that my data use was notably constrained when I revisited this project in the last few months; it's lamentable that Google put such a harsh limitation on API quotas. That being said, I agree that the caching feature you describe is a top priority, but it's unfortunately a tad clumsy to add to this project. What I mean by that is that it would be best to package such a feature as some form of daemon-friendly script, but that's a bit tricky to do with a purely Javascript implementation (that is, without getting something terribly heavyweight like `node.js` involved) and that leaves the options of duplicating the queries in a different language or requiring users to manually submit caching requests every time their quota refills. I'm certainly open to other suggestions if you have them, but I've been hesitant to pursue either of the aforementioned options because they both seem too fragile.
